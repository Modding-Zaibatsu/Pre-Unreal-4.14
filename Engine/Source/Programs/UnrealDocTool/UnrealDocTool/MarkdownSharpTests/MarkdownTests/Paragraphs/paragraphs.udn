A paragraph is simply one or more consecutive lines of text, separated by one
or more blank lines. (A blank line is any line that looks like a blank line — a
line containing nothing but spaces or tabs is considered blank.) Normal
paragraphs should not be indented with spaces or tabs.

The implication of the “one or more consecutive lines of text” rule is that
Markdown supports “hard-wrapped” text paragraphs. This differs significantly
from most other text-to-HTML formatters (including Movable Type’s “Convert Line
Breaks” option) which translate every line break character in a paragraph into
a &lt;br /&gt; tag.

When you do want to insert a &lt;br /&gt; break tag using Markdown, you end a line
with two or more spaces, then type return.

Yes, this takes a tad more effort to create a &lt;br /&gt;, but a simplistic “every
line break is a &lt;br /&gt;” rule wouldn’t work for Markdown. Markdown’s email-style
blockquoting and multi-paragraph list items work best — and look better — when
you format them with hard breaks.