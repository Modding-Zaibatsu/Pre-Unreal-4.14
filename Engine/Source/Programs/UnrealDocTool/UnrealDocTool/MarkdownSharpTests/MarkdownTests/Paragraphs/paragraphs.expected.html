<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" dir="ltr" lang="en-US">
	<head>
	</head>
	<body>
		<p>
			A paragraph is simply one or more consecutive lines of text, separated by one
			or more blank lines. (A blank line is any line that looks like a blank line — a
			line containing nothing but spaces or tabs is considered blank.) Normal
			paragraphs should not be indented with spaces or tabs.
		</p>

		<p>
			The implication of the “one or more consecutive lines of text” rule is that
			Markdown supports “hard-wrapped” text paragraphs. This differs significantly
			from most other text-to-HTML formatters (including Movable Type’s “Convert Line
			Breaks” option) which translate every line break character in a paragraph into
			a &lt;br /&gt; tag.
		</p>

		<p>
			When you do want to insert a &lt;br /&gt; break tag using Markdown, you end a line
		    with two or more spaces, then type return.
		</p>

	    <p>
	        Yes, this takes a tad more effort to create a &lt;br /&gt;, but a simplistic “every
	        line break is a &lt;br /&gt;” rule wouldn’t work for Markdown. Markdown’s email-style
	        blockquoting and multi-paragraph list items work best — and look better — when
	        you format them with hard breaks.
	    </p>
	</body>
</html>
